dspace-spring-rest/src/main/java/org/dspace/app/rest/converter/WorkflowItemConverter.java

```diff
@@ -0,0 +1,161 @@
+        Item item = obj.getItem();
+        EPerson submitter = null;
+        try {
+            submitter = obj.getSubmitter();
+        } catch (SQLException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+
+        witem.setId(obj.getID());
+        witem.setCollection(collection != null ? collectionConverter.convert(collection) : null);
+        witem.setItem(itemConverter.convert(item));
+        witem.setSubmitter(epersonConverter.convert(submitter));
+
+        // 1. retrieve the submission definition
+        // 2. iterate over the submission section to allow to plugin additional
+        // info
+
+        if (collection != null) {
+            SubmissionDefinitionRest def = submissionDefinitionConverter
+                .convert(submissionConfigReader.getSubmissionConfigByCollection(collection.getHandle()));
+            witem.setSubmissionDefinition(def);
+            for (SubmissionSectionRest sections : def.getPanels()) {
+                SubmissionStepConfig stepConfig = submissionSectionConverter.toModel(sections);
+
+                /*
+                 * First, load the step processing class (using the current
+                 * class loader)
+                 */
+                ClassLoader loader = this.getClass().getClassLoader();
+                Class stepClass;
+                try {
+                    stepClass = loader.loadClass(stepConfig.getProcessingClassName());
+
+                    Object stepInstance = stepClass.newInstance();
+
+                    if (stepInstance instanceof AbstractRestProcessingStep) {
+                        // load the interface for this step
+                        AbstractRestProcessingStep stepProcessing =
+                            (AbstractRestProcessingStep) stepClass.newInstance();
+                        for (ErrorRest error : stepProcessing.validate(submissionService, obj, stepConfig)) {
+                            addError(witem.getErrors(), error);
+                        }
+                        witem.getSections()
+                            .put(sections.getId(), stepProcessing.getData(submissionService, obj, stepConfig));
+                    } else {
+                        log.warn("The submission step class specified by '" + stepConfig.getProcessingClassName() +
+                                 "' does not extend the class org.dspace.app.rest.submit.AbstractRestProcessingStep!" +
+                                 " Therefore it cannot be used by the Configurable Submission as the " +
+                                 "<processing-class>!");
+                    }
+
+                } catch (Exception e) {
+                    log.error(e.getMessage(), e);
+                }
+
+            }
+        }
+        return witem;
+    }
+
+    @Override
+    public XmlWorkflowItem toModel(WorkflowItemRest obj) {
+        return null;
+    }
+
+
+    public void addError(List<ErrorRest> errors, ErrorRest toAdd) {
```

Comment:
Please add Javadocs

Reply:
it is (now) a private method https://github.com/DSpace/DSpace/pull/2312/files#diff-6e307cdd13b1a67847458f429e6bb747R139